Sources/GoogleAI/GenerativeModel.swift

```diff
@@ -325,9 +331,19 @@ public final class GenerativeModel {
   public func countTokens(_ content: @autoclosure () throws -> [ModelContent]) async throws
     -> CountTokensResponse {
     do {
-      let countTokensRequest = try CountTokensRequest(
+      let generateContentRequest = try GenerateContentRequest(model: modelResourceName,
```

Comment:
Did you consider passing `nil` for model in this single usage (in lieu of the added `isModelEncoded`)?

Reply:
Unfortunately this isn't possible since, despite not being encoded, `model` is used in the [request URL](https://github.com/google-gemini/generative-ai-swift/blob/edc9de31896e999eb27d636a946ca69428265f3f/Sources/GoogleAI/GenerateContentRequest.swift#L47-L48) when performing `generateContent` and `generateContentStream`.